Pcap++/src/DpdkDevice.cpp

```diff
@@ -518,7 +526,12 @@ void DpdkDevice::getLinkStatus(LinkStatus& linkStatus) const
 	rte_eth_link_get((uint8_t) m_Id, &link);
 	linkStatus.linkUp = link.link_status;
 	linkStatus.linkSpeedMbps = (unsigned) link.link_speed;
+
+#if (RTE_VER_YEAR < 22) || (RTE_VER_YEAR == 22 && RTE_VER_MONTH < 11)
 	linkStatus.linkDuplex = (link.link_duplex == ETH_LINK_FULL_DUPLEX) ? LinkStatus::FULL_DUPLEX : LinkStatus::HALF_DUPLEX;
+#else
+	linkStatus.linkDuplex = (link.link_duplex == RTE_ETH_LINK_FULL_DUPLEX) ? LinkStatus::FULL_DUPLEX : LinkStatus::HALF_DUPLEX;
```

Comment:
For all these consts that have changes in newer DPDK versions, we usually a "PcapPlusPlus" version of them once at the beginning of `DpdkDevice.cpp` which might change per version, like this: https://github.com/seladb/PcapPlusPlus/blob/master/Pcap%2B%2B/src/DpdkDevice.cpp#L52-L58

Then we don't have to do these `#if..#else` everywhere, and instead we just used our own const.

As an example: we define `DPDK_CONFIG_MQ_RSS` which can be `ETH_RSS` in older DPDK versions or `RTE_ETH_MQ_RX_RSS` in newer DPDK versions. In the code we use  `DPDK_CONFIG_MQ_RSS` everywhere